Entities/ACrab.cpp

```diff
@@ -2181,23 +2187,19 @@ void ACrab::Update()
                 m_JetTimeLeft = m_JetTimeTotal;
         }
 
+		float maxAngle = c_HalfPI * m_JetAngleRange;
         // Direct the jetpack nozzle according to movement stick if analog input is present
         if (m_Controller.GetAnalogMove().GetMagnitude() > 0.1)
         {
-            float jetAngle = m_Controller.GetAnalogMove().GetAbsRadAngle() + c_PI;
-            // Clamp the angle to 45 degrees down cone with centr straight down on body
-            if (jetAngle > c_PI + c_HalfPI + c_QuarterPI)// - c_SixteenthPI)
-                jetAngle = c_PI + c_HalfPI + c_QuarterPI;// - c_SixteenthPI;
-            else if (jetAngle < c_PI + c_QuarterPI)// + c_SixteenthPI)
-                jetAngle = c_PI + c_QuarterPI;// + c_SixteenthPI;
-
+			float jetAngle = max(min(m_Controller.GetAnalogMove().GetAbsRadAngle() - c_HalfPI, maxAngle), -maxAngle);
```

Comment:
The first half pi subtraction is to re-orient the upwards facing (analog jump) angle down to zero from which positive and negative max angle are measured and clamped, alternatively I could have included the re-orientation in the clamp values like `-maxAngle - c_HalfPI` and `maxAngle - c_HalfPI ` which just seemed uglier to me. I guess I'll leave a comment explaining all of this :p

The latter half pi substraction is to point the jet angle downwards from the previous re-oriented zero position.

Reply:
Ahh, that makes total sense with the explanation, thanks. I think my brain just kinda didn't wanna deal with thinking about what was actually going on haha.
As for where to include the halfPI, I think the best way to do it would be to have a minAngle and a maxAngle, and include it in their initial calculations. One extra line of code for a lot more cleanness and no need for a comment at all.